### PR TITLE
IpcLogger to help with using IPC metrics

### PR DESCRIPTION
**Pull request by brharrington**
Initial version of some basic helpers to aid in using the
common IPC metrics in a consistent manner. As an example
of using the a simple HTTP client based on the java
HttpURLConnection is instrumented. This will replace
the version from the sandbox lib.

### CODE REVIEW

File: spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcLogEntry.java [hunk 1]
```diff
@@ -0,0 +1,1052 @@
...
+          case '\n':
+            builder.append("\\n");
+            break;
+          case '\r':
+            builder.append("\\r");
+            break;
+          case '\t':
+            builder.append("\\t");
+            break;
+          case ' ':
```
Comment (dmuino) [inline spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcLogEntry.java]: Why is this needed?

Comment (brharrington) [reply spectator-ext-ipc/src/main/java/com/netflix/spectator/ipc/IpcLogEntry.java]: It shouldn't be, I think that was a hold over from a previous iteration where the whitespace would match the check in the default block.
